geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionDataStore.java

```diff
@@ -265,6 +265,36 @@ public void visit(Integer bucketId, Region r) {
     return numPrimaries.get();
   }
 
+  public boolean isPartitionedRegionReady(PartitionedRegion partitionedRegion, final int bucketId) {
+    List<PartitionedRegion> colocatedWithList = getColocatedChildRegions(partitionedRegion);
+    if (colocatedWithList.size() == 0) {
+      return partitionedRegion.isInitialized();
+    }
+    return areAllColocatedPartitionedRegionsReady(bucketId, colocatedWithList);
+  }
+
+  private boolean areAllColocatedPartitionedRegionsReady(int bucketId,
+      List<PartitionedRegion> colocatedWithList) {
+    return !colocatedWithList.stream()
+        .filter(partitionedRegion -> isColocatedPartitionedRegionInitialized(partitionedRegion,
+            bucketId) == false)
+        .findFirst().isPresent();
+  }
+
+  private boolean isColocatedPartitionedRegionInitialized(PartitionedRegion partitionedRegion,
+      final int bucketId) {
+    if (!partitionedRegion.isInitialized()
```

Comment:
instead of if (!A || !B) return false
this might be easier to understand:
if (!A) return false
if (!B) return false